### PR TITLE
Add examples E2E job to source control.

### PR DESCRIPTION
**Pull request by spxtr**
ref #18122

Now someone just needs to make it work :P

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -62,6 +62,9 @@
         - 'gce-flannel':
             description: 'Run E2E tests on GCE using Flannel and the latest successful build. This suite is quarantined in a dedicated project because Flannel integration is experimental.'
             timeout: 90
+        - 'gce-examples':
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: we may not want this to email oncall until it's fixed.

or maybe we do? :)

Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: It's currently set to email $DEFAULT_RECIPIENTS, so this doesn't change anything.

I think I remember it used to email just you. Should I set you as the job owner and have it email you?

Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-e2e.yaml]: if it's already emailing $DEFAULT_RECIPIENTS then this seems fine to me.

**Comment by ixdy** [comment]
LGTM

@k8s-bot unit test this please

**Pull request merged**